astropy/timeseries/periodograms/lombscargle/implementations/main.py

```diff
@@ -84,6 +91,8 @@ def validate_method(method, dy, fit_mean, nterms, frequency, assume_regular_freq
     prefer_fast = len(frequency) > 200 and (
         assume_regular_frequency or _is_regular(frequency)
     )
+    # Note that this condition may be simplified (remove `and not fit_mean`)
+    # once fit_mean=True is made the default value for method='scipy'
     prefer_scipy = "scipy" in methods and dy is None and not fit_mean
```

Comment:
I think we should be able to do this now: e.g.
```python
prefer_scipy = "scipy" in methods and dy is None and not (SCIPY_LT_1_15 or fit_mean)
```
So long as we handle `fit_mean` appropriately at the place where this is called.

Reply:
good call. This makes the comment redundant and the simplification more obvious when the time comes !